imod/tests/test_mf6/test_mf6_rch.py

```diff
@@ -324,3 +326,118 @@ def test_clip_box(rch_dict):
+    assert is_planar_grid(data["rch"]["rate"])
+
+    # Act
+    rch = imod.mf6.Recharge.from_imod5_data(data, target_grid)
+
+    # Assert
+    rendered_rch = rch.render(tmp_path, "rch", None, None)
+    assert "maxbound 2162" in rendered_rch
+    assert rendered_rch.count("begin period") == 1
+    assert np.all(
+        rch.dataset["rate"].isel(layer=0).values == data["rch"]["rate"].values
+    )
+
+
+@pytest.mark.usefixtures("imod5_dataset")
+def test_planar_rch_from_imod5_transient(imod5_dataset, tmp_path):
+    data = deepcopy(imod5_dataset)
+    target_grid = data["khv"]["kh"]
+
+    # create a grid with recharge for 3 timesteps
+    input_recharge = data["rch"]["rate"]
+    input_recharge.values[0] = 0
+    input_recharge = input_recharge.expand_dims({"time": [0, 1, 2]})
+
+    # make it planar by setting the layer coordinate to 0
+    input_recharge["layer"].values[0] = 0
+
+    # update the data set
+    data["rch"]["rate"] = input_recharge
+    assert is_transient_data_grid(data["rch"]["rate"])
+    assert is_planar_grid(data["rch"]["rate"])
+
+    # act
+    rch = imod.mf6.Recharge.from_imod5_data(data, target_grid)
+
+    # assert
+    rendered_rch = rch.render(tmp_path, "rch", [0, 1, 2], None)
+    assert rendered_rch.count("begin period") == 3
+    assert "maxbound 2162" in rendered_rch
+    assert np.all(
+        rch.dataset["rate"].isel(layer=0).values == data["rch"]["rate"].values
+    )
+
+
+@pytest.mark.usefixtures("imod5_dataset")
+def test_non_planar_rch_from_imod5_constant(imod5_dataset, tmp_path):
+    data = deepcopy(imod5_dataset)
+
+    # make the first layer of the target grid inactive
+    target_grid = data["khv"]["kh"]
+    target_grid.loc[{"layer": 1}] = 0
+
+    # the input for recharge is on the second layer of the targetgrid
+    input_recharge = nan_like(target_grid)
+    input_recharge.loc[{"layer": 2}] = data["rch"]["rate"].isel(layer=0)
+
+    # update the data set
+    data["rch"]["rate"] = input_recharge
+    assert not is_planar_grid(data["rch"]["rate"])
+    assert not is_transient_data_grid(data["rch"]["rate"])
+
+    # act
+    rch = imod.mf6.Recharge.from_imod5_data(data, target_grid)
+
+    # assert
+    rendered_rch = rch.render(tmp_path, "rch", None, None)
+    assert rendered_rch.count("begin period") == 1
+    assert "maxbound 2162" in rendered_rch
+    assert np.all(
+        rch.dataset["rate"].sel(layer=2).values
+        == data["rch"]["rate"].sel(layer=2).values
+    )
+
+
+@pytest.mark.usefixtures("imod5_dataset")
+def test_non_planar_rch_from_imod5_transient(imod5_dataset, tmp_path):
+    data = deepcopy(imod5_dataset)
+
+    # make the first layer of the target grid inactive
+    target_grid = data["khv"]["kh"]
+    target_grid.loc[{"layer": 1}] = 0
+
+    # the input for recharge is on the second layer of the targetgrid
+    input_recharge = nan_like(target_grid)
+    input_recharge.loc[{"layer": 2}] = data["rch"]["rate"].sel(
+        layer=1, method="nearest"
```

Comment:
Please add some comment why you had to choose "nearest" here (or that you don't know but it was required for some reason)

Reply:
it turned out (as you suggested) that the deepcopy statements at the beginning of these tests ( top copy the fixture) does not work as expected: the fixture data is still modified by the tests. This is why this test needed the method=nearest to succeed when the test was performed along with the other tests. One of the earlier tests tampered with the layer coordinate. I have solved this now by putting the layer coordinate back to its original value in this test. But we should make a true deepcopy of the test data at some point.